utils/record/record.go

```diff
@@ -278,17 +278,29 @@ func (e *EventRecorderAdapter) sendNotifications(object runtime.Object, opts Eve
 		return err
 	}
 
-	res, err := notificationsAPI.RunTrigger(trigger, objMap)
-	if err != nil {
-		log.Errorf("Failed to execute condition of trigger %s: %v", trigger, err)
-		return err
-	}
-	log.Infof("Trigger %s result: %v", trigger, res)
-
-	for _, dest := range destinations {
-		for _, c := range res {
-			if c.Triggered == true {
-				err = notificationsAPI.Send(objMap, triggerActions[0].Send, dest)
+	for _, ta := range triggerActions {
+		if ta.When != "" {
+			res, err := notificationsAPI.RunTrigger(trigger, objMap)
+			if err != nil {
+				log.Errorf("Failed to execute condition of trigger %s: %v", trigger, err)
+				return err
+			}
+			log.Infof("Trigger %s result: %v", trigger, res)
+
+			for _, dest := range destinations {
+				for _, c := range res {
+					if c.Triggered == true {
```

Comment:
Im missing where is the connection between the RunTrigger results (`res`) and the current triggerAction (`ta`). It seems that if `res` has multiple entries, it will send duplicated notifications. Can you please confirm this logic?

Reply:
`Runtrigger` is used only for validating when condition.. if the condition doesnot match it returns with `res` with `Triggered` in it as false and that will not send notification. `res` here holds the template that needs to be sent where as trigger actions holds the condition.

beyond that the logic of execution is same as earlier.

Reply:
@RaviHari Ok. 
But what should happen if `res` has 2 entries: one with `Triggered == false` and the other with `Triggered == true`?
It seems to me that it should only send the notification if all entries are `true` isn't it?

Reply:
ok so I validated it with multiple when conditions and a corresponding send template for each condition..
Then the result object has twice as many entries/elements in the list that of the conditions.. and the result object is going awry.. and the send notification fails.

<img width="907" alt="trigger_action" src="https://user-images.githubusercontent.com/7333720/189686517-6e0973e5-ca20-49dd-9653-51008f7dbe51.png">
<img width="974" alt="Res1" src="https://user-images.githubusercontent.com/7333720/189686564-5e65129a-5685-4835-a4b4-a36d46306c19.png">
<img width="769" alt="Res2" src="https://user-images.githubusercontent.com/7333720/189686555-1c0f290e-81ef-4725-b041-8fe90890c21a.png">
<img width="989" alt="Res3" src="https://user-images.githubusercontent.com/7333720/189686548-92c2ed61-af12-43b9-aeb2-d48a4429fac0.png">
<img width="1000" alt="Res4" src="https://user-images.githubusercontent.com/7333720/189686536-f5bd8e4a-4148-45bf-9697-02b4eb01a5df.png">

as it sees the third element in the list is true it tries to send notification but fails with:
```
{"ok":false,"error":"no_text"}  service=slack
ERRO[0504] notification error: no_text                  
DEBU[0504] Notification sent                             namespace=rollout-canary rollout=rollouts-demo time_ms=439906.596329
```
as the template for that entry is nil.  I will need to discuss this further to see the way forward.